test-case/check-volume-levels.sh

```diff
@@ -0,0 +1,394 @@
+}
+
+# 0..1s channels 1-4 muted
+# 1..2s channels 1-4  nominal gain
+# 2..3s different mute switches
+# 3..4s channels 1-4 nominal gain
+run_test_3 () {
+    arecord_in_background "$ARECORD_WAV3" "$CAP_CHANNELS" 4
+    PID=$?
+    wait
+
+    cset_nom
+    wait
+
+    cset_mute_diff2 "$CAP_CHANNELS"
+    wait
+
+    cset_unmute
+
+    wait_for_pid "$PID"
+    dlogi Ready.
+}
+
+#
+# Helper functions
+#
+
+cset_nom () {
+    dlogi Set nominal volume
+    amixer cset name="$CAP_VOLUME" $VOL_NOM
+}
+
+cset_max () {
+    dlogi Set maximum volume
+    amixer cset name="$CAP_VOLUME" $VOL_MAX
+}
+
+cset_min () {
+    dlogi Set minimum volume
+    amixer cset name="$CAP_VOLUME" $VOL_MIN
+}
+
+cset_mutevol () {
+    dlogi Set mute via volume
+    amixer cset name="$CAP_VOLUME" $VOL_MUTE
+}
+cset_mute () {
+    dlogi Set mute via switch
+    amixer cset name="$CAP_SWITCH" off
+}
+
+cset_unmute () {
+    dlogi Set unmute
+    amixer cset name="$CAP_SWITCH" on
+}
+
+wait () {
+    dlogi Waiting
+    sleep 1
+}
+
+cset_mute_diff1 () {
+    dlogi Set switch pattern 1
+    case "$1" in
+	4)
+	    amixer cset name="$CAP_SWITCH" off,on,on,off
+	    ;;
+	2)
+	    amixer cset name="$CAP_SWITCH" off,on
+	    ;;
+	1)
+	    amixer cset name="$CAP_SWITCH" off
+	    ;;
+    esac
+}
+
+cset_mute_diff2 () {
+    dlogi Set switch pattern 2
+    case "$1" in
+	4)
+	    amixer cset name="$CAP_SWITCH" on,off,off,on
+	    ;;
+	2)
+	    amixer cset name="$CAP_SWITCH" on,off
+	    ;;
+	1)
+	    amixer cset name="$CAP_SWITCH" on
+	    ;;
+    esac
+}
+
+cset_volume_diff1 () {
+    dlogi Set volume pattern 1
+    case "$1" in
+	4)
+	    amixer cset name="$CAP_VOLUME" $VOL_P10DB,$VOL_0DB,$VOL_M10DB,$VOL_M30DB
+	    ;;
+	2)
+	    amixer cset name="$CAP_VOLUME" $VOL_P10DB,$VOL_0DB
+	    ;;
+	1)
+	    amixer cset name="$CAP_VOLUME" $VOL_P10DB
+	    ;;
+    esac
+}
+
+cset_volume_diff2 () {
+    dlogi Set volume pattern 2
+    case "$1" in
+	4)
+	    amixer cset name="$CAP_VOLUME" $VOL_M10DB,$VOL_P10DB,$VOL_0DB,$VOL_M20DB
+	    ;;
+	2)
+	    amixer cset name="$CAP_VOLUME" $VOL_M10DB,$VOL_P10DB
+	    ;;
+	1)
+	    amixer cset name="$CAP_VOLUME" $VOL_M10DB
+	    ;;
+    esac
+}
+
+wait_for_pid () {
+    dlogi Waiting for process "$1" to complete
+    tail --pid="$pid" -f /dev/null
+}
+
+arecord_in_background () {
+    dlogi Capturing file "$1"
+    arecord -D "$CAP_HW" -f $CAP_FORMAT -r $CAP_RATE -c "$2" -d "$3" "$1"  2> /dev/null &
+    pid=$!
+    return $pid
+}
+
+aplay_in_background () {
+    if ! [ -f "$1" ]; then
+	dloge "Error: File $1 does not exist"
+	exit 1
+    fi
+
+    dlogi Playing file "$1"
+    aplay -D "$PLAY_HW" "$1"  2> /dev/null &
+    pid=$!
+    return $pid
+}
+
+do_preparations () {
+
+    # Check that TPLG is set
+    if [ -z "$TPLG" ]; then
+	dloge "Error: Need to set environment variable TPLG to the used SOF topology"
+	exit 1
+    fi
+
+    if [[ "$TPLG" != *"nocodec"* ]]; then
+	dlogi "Note: this test is executed only with nocodec topologies."
+	exit 0
```

Comment:
There is a special exit code that Jenkins interprets as "SKIP", I think it's 2 but please double check.

Reply:
Good idea. Though 2 seems to be unstable. I'll search more.

Reply:
I saw in another comment git commit text that mentioned value 2 as Non Applicable. I'll use that for now.